general-concepts/dependencies/text.xml

```diff
@@ -850,5 +850,309 @@ USE flags. This would then break building your ebuild.
+# Bad (if app-misc/foo is slotted):
+DEPEND="&gt;=app-misc/foo-2
+	&lt;app-misc/foo-5"
+</codesample>
+
+<p>
+If <c>app-misc/foo</c> is slotted and two installed slots each satisfy one
+side of the range, the resolver can legally use them separately. Likewise,
+independent USE requirements:
+</p>
+
+<codesample lang="ebuild">
+# Bad (if app-misc/foo is slotted):
+RDEPEND="app-misc/foo[foo]
+	bar? ( app-misc/foo[baz] )"
+</codesample>
+
+<p>
+may be met by different slots with different USE configurations. For
+packages that are not truly multi-slotted (i.e. slots represent mutually
+incompatible ABIs), the safe fix is to state the intended slot explicitly or
+to express the intent via any-of groups limited per slot.
+</p>
+
+<p>
+For a version range across two slots of <c>sys-libs/db</c>, prefer:
+</p>
+
+<codesample lang="ebuild">
+# Good:
+DEPEND="
+	|| (
+		=sys-libs/db-5*:5
+		=sys-libs/db-4*:4
+	)
+"
+</codesample>
+
+<p>
+If you must combine different USE requirements, nest them per slot:
+</p>
+
+<codesample lang="ebuild">
+# Good:
+RDEPEND="
+	|| (
+		( sys-libs/db:5 tools? ( sys-libs/db:5[cxx] ) )
+		( sys-libs/db:4 tools? ( sys-libs/db:4[cxx] ) )
+	)
+"
+</codesample>
+
+</body>
+</subsection>
+
+<subsection>
+<title>The <c>:=</c> slot operator with multiple slots</title>
+<body>
+
+<p>
+The <c>:=</c> operator records the slot/sub-slot of the best matching
+installed version for the given atom (see PMS). If that atom can match
+newer slots than the versions you explicitly allowed elsewhere, it may bind
+to the wrong slot (and even pull it in during build).
+</p>
+
+<p>
+To prevent this, ensure the atom carrying <c>:=</c> cannot match newer slots
+than intended. One simple pattern is to cap it with an upper bound that
+excludes unwanted major versions while keeping <c>:=</c> on the same
+package:
+</p>
+
+<codesample lang="ebuild">
+# Good:
+DEPEND="
+	|| (
+		=sys-libs/db-5*
+		=sys-libs/db-4*
+	)
+	&lt;sys-libs/db-6:=
+"
+</codesample>
+
+<p>
+This forces the slot-operator binding to a version in the requested range at
+build time. Be cautious when combining multiple conditional USE sets with a
+slot operator; keeping the <c>:=</c> atom simple and separately constrained
+is usually clearer.
+</p>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Understanding any-of dependencies</title>
+<body>
+
+<p>
+An any-of group (<c>|| ( ... )</c>) guarantees only that at least one
+immediate child element is satisfied for the relevant dependency class. It
+does not guarantee which element will be chosen, nor does it bind the choice
+made at build time to the choice used later at runtime. Order can be used as
+an implementation hint but is not a contract.
+</p>
+
+<codesample lang="ebuild">
+DEPEND="
+	|| (
+		dev-libs/A
+		dev-libs/B
+		dev-libs/C
+	)
+"
+</codesample>
+
+<p>
+If more than one alternative is installed, the package manager may use any
+of them. Replacing one alternative with another later still satisfies the
```

Comment:
I would be careful with "package manager using" anything, given that the package manager doesn't really do anything here, beyond making sure that at least one of the atoms is satisfied. It's the ebuild / package code itself deciding which will be used.